src/pages/Florist/FloristWithCart/FloristWithCart.tsx

```diff
@@ -0,0 +1,201 @@
+import {
+  Button,
+  createStyles,
+  Fab,
+  Grid,
+  makeStyles,
+  Menu,
+  MenuItem,
+  Theme,
+  Typography,
+} from "@material-ui/core";
+import { useEffect, useMemo, useRef, useState } from "react";
+import ShoppingCartIcon from "@material-ui/icons/ShoppingCart";
+import BackspaceIcon from "@material-ui/icons/Backspace";
+import styles from "./Florist.module.css";
+import { Flower } from "../../../models/flower.model";
+import { getFlowers } from "../../../helpers/api/flowers/getFlowers";
+import { useApi } from "../../../utils/api/useApi";
+import { LoadingWrapper } from "../../../components/LoadingWrapper";
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    container: {
+      width: "90%",
+      marginLeft: "2.5%",
+      padding: "5%",
+    },
+    header: {
+      margin: "2%",
+      marginLeft: "5%",
+      width: "100%",
+    },
+    flowerList: {
+      width: "100%",
+    },
+    card: {
+      margin: "1%",
+      height: "30%",
+      width: "30%",
+      display: "flex",
+      flexDirection: "column",
+      alignItems: "center",
+    },
+    selected: {
+      backgroundColor: "#6ac69780",
+    },
+    notSelected: {
+      backgroundColor: theme.palette.background.default,
+    },
+    cartButton: {
+      marginLeft: "0%",
+      left: "15%",
+    },
+    deleteButton: {
+      color: theme.palette.error.main,
+      padding: "5%",
+    },
+    buyFlowers: {
+      backgroundColor: theme.palette.primary.main,
+      color: theme.palette.background.paper,
+      "&:hover": {
+        backgroundColor: theme.palette.primary.dark,
+      },
+    },
+  })
+);
+
+export const Florist = () => {
+  const [flowersAPIState, getAllFlowers] = useApi(getFlowers);
+  const allFlowers = useMemo(
+    () => flowersAPIState.response ?? [],
+    [flowersAPIState]
+  );
+  const calledOnMount = useRef(false);
+  useEffect(() => {
+    if (!calledOnMount.current) {
+      getAllFlowers();
+    }
+    calledOnMount.current = true;
+  }, [getAllFlowers]);
+
+  const [cart, setCart] = useState<Flower[]>([]);
```

Comment:
This is the previous version for buying multiple flowers at a time with a cart, currently unused.

Reply:
Could you rename the component and add a comment